packages/fxa-auth-server/lib/payments/capability.ts

```diff
@@ -845,6 +888,36 @@ export class CapabilityService {
    * Retrieve the client capabilities
    */
   async getClients() {
+    const emptyClients: {
+      clientId: string;
+      capabilities: string[];
+    }[] = [];
+    const contentfulEnabled = config.getProperties().contentful.enabled;
+
+    if (contentfulEnabled) {
+      if (!this.capabilityManager) {
+        if (this.logToSentry('getClients.CapabilityManagerNotFound')) {
+          Sentry.captureMessage(
+            `CapabilityManager not found.`,
+            'error' as SeverityLevel
+          );
+        }
+
+        return emptyClients;
```

Comment:
With Contentful enabled, If an error occurred this method should reject with the error, instead of returning `emptyClients`. 

Reason for that is that if there is an error, consumers of the API should be made aware there's an error. By returning `emptyClients`, consumers will get a status code `200`, indicating success, and assume that the result was `emptyClients`. Whereas if the API returns with status code `500` and the error, they'll know something's gone wrong, and can handle it appropriately. (Also at this point, we'll know somethings gone wrong)

(This comment applies to other instances where returning `emptyClients` and `emptyCapabilities`.)

Reply:
Looks like with the latest changes this comment still applies. As an example, the [try...catch on lines 344 - 355](https://github.com/mozilla/fxa/blob/9aa8e4d78bbc884c53ec5ca982ddb19119d208cd/packages/fxa-auth-server/lib/payments/capability.ts#L344-L355) will catch the error and then continue on, instead of rejecting with the error.

Reply:
Updated to a `throw`. Let me know if this is what you meant or if you had a different idea.

Reply:
Yup `throw` is right. Note that this applies generally for changes in this PR, where it makes sense, not just this line. 

(Regarding, "where it makes sense", see the original note about wanting to reject with an error so that consumers of these methods are aware that an error occurred. Let me know if you'd like to discuss more.)

The function signature for `AppError.internalValidationError` is also a little different than how it's being used at the moment. `AppError.internvalValidationError = (op, data, error) => { ... }`, so you'll just have to update that as well to pass the error at the expected place.

Optionally, could keep the `log.error` in addition to throwing the error, just for some additional logging.